test/wdio/attribute-host/cmp.test.tsx

```diff
@@ -0,0 +1,103 @@
+import { h } from '@stencil/core';
+import { render } from '@wdio/browser-runner/stencil';
+
+describe('attribute-host', function () {
+  before(async () => {
+    render({
+      template: () => <attribute-host></attribute-host>,
+    });
+  });
+
+  it('add/remove attrs', async () => {
+    await $('section').waitForExist();
+
+    const elm = await $('section');
+    const button = await $('button');
+
+    await expect(elm).toHaveAttribute('content', 'attributes removed');
+    await expect(elm).toHaveAttribute('bold', 'false');
+    await expect(elm).not.toHaveAttribute('padding');
+    await expect(elm).not.toHaveAttribute('margin');
+    await expect(elm).not.toHaveAttribute('color');
+    await expect(elm).not.toHaveAttribute('no-attr');
+
+    let defaultBorderColor = getComputedStyle(document.body).borderColor.replaceAll(' ', '');
+    if (browser.isChromium) {
+      defaultBorderColor = defaultBorderColor.replace('rgb(', 'rgba(').replace(')', ',1)');
+    }
+    await expect(elm).toHaveStyle({
+      // get default border color from body element as it might differ between different OS
+      'border-color': defaultBorderColor,
```

Comment:
This is unfortunately a limitation within [the spec](https://github.com/w3c/webdriver/issues/1447) which hasn't been addressed yet. I raised a PR to the wpt.fyi to push on this topic. Anyway for now we have to do some browser checks to validate the right value.

The reason why I am doing `getComputedStyle(document.body).borderColor.replaceAll(' ', '')` is because we are validating the "before" state here which refers to the system default color for this HTML element. Running this in CI has shown that this default color differs between Mac (my machine) and Linux (the machine we use in CI).